src/SFA.DAS.Forecasting.Application/Estimations/Validation/AddApprenticeshipValidator.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.Forecasting.Domain.Shared.Validation;
+using SFA.DAS.Forecasting.Models.Estimation;
+
+namespace SFA.DAS.Forecasting.Application.Estimations.Validation
+{
+    public class AddApprenticeshipValidator : IAddApprenticeshipValidator
+    {
+        public List<ValidationResult> ValidateApprenticeship(ApprenticeshipToAdd apprenticeshipToAdd)
+        {
+            var validationResults = new List<ValidationResult>
+            {
+                ApprenticeshipSelected(apprenticeshipToAdd),
+                NumberOfApprenticeshipsEntered(apprenticeshipToAdd),
+                NumberOMonthsEntered(apprenticeshipToAdd),
+                NumberOfMonthsAcceptable(apprenticeshipToAdd),
+                StartMonthEntered(apprenticeshipToAdd),
+                StartYearEntered(apprenticeshipToAdd),
+                StartDateNotInPast(apprenticeshipToAdd),
+                StartDateNotInFarFuture(apprenticeshipToAdd),
+                WithinTotalCap(apprenticeshipToAdd),
+                TotalCapEntered(apprenticeshipToAdd)
+            };
+
+            return validationResults.Where(res => res.IsValid == false).ToList();
+        }
+
+        private static ValidationResult ApprenticeshipSelected(ApprenticeshipToAdd apprenticeshipToAdd)
+        {
+            return apprenticeshipToAdd.CourseId == null
+                ? ValidationResult.Failed("NoApprenticeshipSelected")
+                : ValidationResult.Success;
+        }
+
+        private static ValidationResult NumberOfApprenticeshipsEntered(ApprenticeshipToAdd apprenticeshipToAdd)
+        {
+            return apprenticeshipToAdd.ApprenticesCount == null || apprenticeshipToAdd.ApprenticesCount <= 0
+                ? ValidationResult.Failed("NoNumberOfApprentices")
+                : ValidationResult.Success;
+        }
+
+        private static ValidationResult NumberOMonthsEntered(ApprenticeshipToAdd apprenticeshipToAdd)
+        {
+            return apprenticeshipToAdd.NumberOfMonths == null ? ValidationResult.Failed("NoNumberOfMonths") : ValidationResult.Success;
+        }
+
+        private static ValidationResult NumberOfMonthsAcceptable(ApprenticeshipToAdd apprenticeshipToAdd)
+        {
+            return apprenticeshipToAdd.NumberOfMonths.HasValue && apprenticeshipToAdd.NumberOfMonths < 12
+                ? ValidationResult.Failed("ShortNumberOfMonths")
+                : ValidationResult.Success;
+        }
+
+        private static ValidationResult StartMonthEntered(ApprenticeshipToAdd apprenticeshipToAdd)
```

Comment:
Looks like you could refactor some of this code and the code in the StartMonthAndStartYearAreAcceptable to reduce the amount of code.